pkg/resources/connection.go

```diff
@@ -0,0 +1,313 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/schemas"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var connectionSchema = map[string]*schema.Schema{
+	"name": {
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		Description:      blocklistedCharactersFieldDescription("String that specifies the identifier (i.e. name) for the connection. Must start with an alphabetic character and may only contain letters, decimal digits (0-9), and underscores (_). For a primary connection, the name must be unique across connection names and account names in the organization. For a secondary connection, the name must match the name of its primary connection."),
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
+	"as_replica_of": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		Description:      "Specifies the identifier for a primary connection from which to create a replica (i.e. a secondary connection).",
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
+	"is_primary": {
+		Type:         schema.TypeBool,
+		Optional:     true,
+		RequiredWith: []string{"as_replica_of"},
+	},
+	"enable_failover_to_accounts": {
+		Type:        schema.TypeList,
+		Optional:    true,
+		Description: "Enables failover for given connection to provided accounts. Specifies a list of accounts in your organization where a secondary connection for this primary connection can be promoted to serve as the primary connection. Include your organization name for each account in the list.",
+		MinItems:    1,
+		Elem:        &schema.Schema{Type: schema.TypeString},
```

Comment:
I think we should have `DiffSuppressFunc: suppressIdentifierQuoting` in elem.

Reply:
+1